### ISSUE TITLE
Undefined offset - line 28

### ISSUE THREAD

**Issue by ghost** [heidilabs/markov-bot#1]
HEY, I have it setup just as you say to, the bot will parse any .txt file I host but nothing off the web, no twitter, no rss feed. This is the traceback.

PHP Notice:  Undefined offset: 1 in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 28
PHP Notice:  Undefined offset: 1 in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 28
PHP Warning:  fopen(/home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/Service/../../../cache/twitter.com/hire_ai-): failed to open stream: No such file or directory in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 50
PHP Warning:  fwrite() expects parameter 1 to be resource, boolean given in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 51
PHP Warning:  fclose() expects parameter 1 to be resource, boolean given in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 52
PHP Notice:  Undefined offset: 1 in /home/valest0rm/bots/setEnv/util/myDirectory/src/MarkovBot/SampleCache.php on line 28

**Comment by ghost** [comment]
you have it set so that you add a twitter as so:
- twitter://erikaheidi

I have also tried adding https://, http://, ://, and /
None of which help the issue.

:(

**Comment by erikaheidi** [comment]
Hmm that's strange... when you use the twitter source, what do you get when you run `$ php console.php twitter:test`  ? should show the twitter user that it's connected.

**Comment by erikaheidi** [comment]
Now looking at the error more closely, on line 28, it seems that it doesn't find the second portion of the path when running `explode`.  The format for Twitter should be just **twitter://username**. Maybe something not correctly idented on the YAML file causing the source to be blank?

**Comment by ghost** [comment]
yes. it was a [tab] instead of space*4 that screwed my config.yml edit up.

but now.

Preparing Tweet...
Account in use: [IT GIVES MY PROPER ACCOUNT] so my api keys r good
Posting: ... [IT DISPLAYS A QUIRKY MARKOV THING]

but no tweet ever appears on my timeline D;

**Comment by ghost** [comment]
rawr! i cant figure it out because everything appears to be working smoothly. i mean it even says the correct twitter name i have the app connected to. and says it posted. but nothing appears on twitter. baffled. id assume if something was failing it'd say so in terminal. and ive gone over your readme 1000x :(

**Comment by erikaheidi** [comment]
hm, there's a chance your app is configured for read only, so it says it worked but it doesn't post anything. can you have a look at your app settings to make sure it's read **\+ write** ?

**Comment by ghost** [comment]
yep. its set to read and write.

on a side note. i wrote this: http://static.inky.ws/syn/1133/5481704131674
its a block from an a.i. script ive been working on, that now incorporates your php work :)
that block runs your console as :test and reads the parsed data in terminal, then uses twitter's command line tool to tweet it, as well as use it as a response to chatter in irc.

**Comment by alyak** [comment]
RE: https://github.com/heidilabs/markov-bot/issues/1#issuecomment-93183207

>http://crunknet.tumblr.com/libaip-rp
;x
